index.html

```diff
@@ -242,6 +242,14 @@ <h1 class="collapse">Discounted Dining Finder: Places to Eat Out To Help Out</h1
         const x = Math.round((+result.lat - 49.0) / (12.0 / 160.0));
         const y = Math.round((+result.lon + 9) / (11.0 / 80.0));
 
+        if (sessionStorage) {
+                old = JSON.parse(sessionStorage.getItem('cachedLocation'));
+                if (old && old.x == x && old.y == y) {
+                        return;
+                }
+                sessionStorage.setItem('cachedLocation', JSON.stringify({"x": x, "y": y}));
+        }
```

Comment:
OBSERVATION: Should be able to get the current location from the location hash in the URL, so wouldn't need session storage.

Reply:
I did have it working from the current URL as thought it would be easier and not storing something is always better than storing something. The issue was coming back to the map after leaving wouldn't refresh the data set. So if something gets added or removed in the meantime the user would never know. Session ties it to the current session so it's cached for as long as the current page lasts.  So time ranges the cache too.